cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h

```diff
@@ -0,0 +1,62 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: December, 2015
+ *   Base class implementation for trajectory_profile_generator.
+ * \brief
+ *
+ *
+ ****************************************************************/
+
+#ifndef COB_CONTROL_COB_CARTESIAN_CONTROLLER_INCLUDE_COB_CARTESIAN_CONTROLLER_TRAJECTORY_PROFILE_GENERATOR_BASE_H_
+#define COB_CONTROL_COB_CARTESIAN_CONTROLLER_INCLUDE_COB_CARTESIAN_CONTROLLER_TRAJECTORY_PROFILE_GENERATOR_BASE_H_
+
+#define MIN_VELOCITY_THRESHOLD 0.001
+
+#include "ros/ros.h"
+#include "cob_cartesian_controller/cartesian_controller_data_types.h"
+#include <cob_cartesian_controller/cartesian_controller_utils.h>
+
+class TrajectoryProfileBase
+{
+    public:
+        TrajectoryProfileBase(const cob_cartesian_controller::CartesianActionStruct& params):
+            params_(params)
+        {}
+
+        virtual ~TrajectoryProfileBase() {}
+
+        virtual cob_cartesian_controller::ProfileTimings getProfileTimings(double Se, double te, double accl, double vel, bool calcMaxTe) = 0;
```

Comment:
in order to get rid of the boolean flag `ok` in the ProfileTimings struct, we could let this function return bool, i.e. `true` for "ok" and false for "not ok", the ProfileTimings would then be passed by reference...

Minor: For me, it's nicer to the eye if `vel` comes before `acc`...

`virtual bool getProfileTimings(double Se, double te, double vel, double accl, bool calcMaxTe, cob_cartesian_controller::ProfileTimings& pt) = 0;`